src/plugins/content/accessibility/keyNavigation/keyNavigation.js

```diff
@@ -0,0 +1,161 @@
+
+    /**
+     * @typedef {Object} testRunnerKeyNavigator
+     */
+    return {
+        /**
+         * Setup the keyNavigator
+         * @returns {testRunnerKeyNavigator}
+         */
+        init() {
+            const navigationMode = modeFactory(contentNavigatorType, config);
+            const navigationConfig = navigationMode.config;
+            strategies = getStrategies(navigationMode, testRunner);
+            const navigators = getNavigators(strategies);
+
+            //blur current focused element, to reinitialize keyboard navigation
+            if (document.activeElement) {
+                document.activeElement.blur();
+            }
+
+            groupNavigator = keyNavigator({
+                id: 'test-runner',
+                replace: true,
+                loop: true,
+                elements: navigableGroupElement.createFromNavigators(navigators),
+                // we don't need to propagate tabs for the main navigation, because we've rewritten them and this is not an element
+                // there is an issue with nested navigators
+                propagateTab: navigationConfig.propagateTab
+            });
+
+            if (navigationConfig.flatNavigation) {
+                navigators.forEach(navigator => {
+                    navigator
+                        .on('upperbound', () => {
+                            if (allowedToNavigateFrom(navigator)) {
+                                groupNavigator.next();
+                            }
+                        })
+                        .on('lowerbound', () => {
+                            if (allowedToNavigateFrom(navigator)) {
+                                groupNavigator.previous();
+                                groupNavigator.getCursor().navigable.getKeyNavigator().last();
+                            }
+                        });
+                });
+            } else {
+                setupItemsNavigator(groupNavigator, {
+                    keyNextItem: navigationConfig.keyNextGroup,
+                    keyPrevItem: navigationConfig.keyPrevGroup
+                });
+            }
+
+            shortcut
+                .remove(eventNS)
+                .add(`tab${eventNS} shift+tab${eventNS}`, function (e) {
+                    if (!allowedToNavigateFrom(e.target)) {
+                        return false;
+                    }
+                    if (!groupNavigator.isFocused()) {
+                        groupNavigator.focus();
+                    }
+                });
+
+            return this;
+        },
+
+        /**
+         * Gets the attached testRunner
+         * @returns {testRunner}
+         */
+        getTestRunner() {
+            return testRunner;
+        },
+
+        /**
+         * Switches the navigation mode
+         * @param {String} newMode
+         * @returns {testRunnerKeyNavigator}
+         */
+        setMode(newMode) {
+            contentNavigatorType = newMode;
```

Comment:
is it enough to switch mode? Or some part of function from `init` should be reinvoke

Reply:
Yes, this is a `soft` change. The strategies will be applied on the next call to `init`.